angular2-jwt.ts

```diff
@@ -15,46 +15,41 @@ export interface IAuthConfig {
   noTokenScheme?:boolean;
 }
 
+class AuthConfigConsts {
+    public static readonly DEFAULT_TOKEN_NAME = 'id_token';
+    public static readonly DEFAULT_HEADER_NAME = 'Authorization';
+    public static readonly HEADER_PREFIX_BEARER = 'Bearer ';
+}
+
 /**
  * Sets up the authentication configuration.
  */
 
 export class AuthConfig {
-  
-  headerName: string;
-  headerPrefix: string;
-  tokenName: string;
-  tokenGetter: any;
-  noJwtError: boolean;
-  noTokenScheme: boolean;
-  globalHeaders: Array<Object>;
 
-  constructor(config:any={}) {
-    this.headerName = config.headerName || 'Authorization';
-    if (config.headerPrefix) {
-      this.headerPrefix = config.headerPrefix + ' ';
-    } else if (config.noTokenScheme) {
-      this.headerPrefix = '';
+  private _config : IAuthConfig;
+
+  constructor(config:IAuthConfig = {
```

Comment:
> You can still have typings for the interface. You just make all the fields optional.

That's what I have proposed in the first paragraph of [this comment](https://github.com/auth0/angular2-jwt/pull/93#discussion_r65080928), but I think we will get in trouble having all these optional fields further down the road, because when calling `getConfig` method, we know that all properties are set and they are not optional anymore.